interfaces/MatlabToolbox/matlab_new/Reactors/Wall.m

```diff
@@ -0,0 +1,252 @@
+classdef Wall < handle
+    
+    properties
+        id
+        type
+        left
+        right
+        area
+    end
+    
+    methods
+        %% Wall class constructor
+
+        function x = Wall(left, right, area, k, u, q, v)
+            % A Wall separates two reactors, or a reactor and a reservoir.
+            % A Wall has a finite area, may conduct heat between the two
+            % reactors on either side, and may move like a piston.
+            %
+            % Walls are stateless objects in Cantera, meaning that no
+            % differential equation is integrated to determine any wall
+            % property. Since it is the wall (piston) velocity that enters
+            % the energy equation, this means that it is the velocity, not
+            % the acceleration or displacement, that is specified. The wall
+            % velocity is computed from:
+            %
+            %    v = K(P_left - P_right) + v_0(t),
+            %
+            % where K is a non-negative constant, and v_0 is a specified
+            % function of time. The velocity is positive if the wall is
+            % moving to the right. 
+            %
+            % The heat flux through the wall is computed from:
+            %
+            %    q = U(T_left - T_right) + q_0(t),
+            %
+            % where U is the overall heat transfer coefficient for
+            % conduction/convection, and q_0 is a specified function of
+            % time. The heat flux is positive when heat flows from the
+            % reactor on the left to the reactor on the right. 
+            %
+            % Note: all of the arguments are optional and can be activated
+            % after initial construction by using the various methods of
+            % the 'Wall' class. Any improperly specified arguments will
+            % generate warnings; these can be ignored if the intention was
+            % to not use a particular argument. Thus, the velocity of the
+            % wall can be set by using empty strings or 0.0 for each of the
+            % arguments before the velocity with no harm. 
```

Comment:
Using a class in its intended way should not produce warnings. I would suggest just removing the ability to specify optional properties as part of the constructor, and rely on setting the ones that have non-default values in subsequent statements.

Reply:
So if I understand correctly, the preferred way would be for the class constructor to only create the object and set default values. Then the user would set those values outside of the class constructor?

Reply:
Yes, that's correct.